src/mono/wasm/runtime/strings.ts

```diff
@@ -30,26 +30,23 @@ export class StringDecoder {
             pLengthBytes = <any>this.mono_wasm_string_decoder_buffer + 4,
             pIsInterned = <any>this.mono_wasm_string_decoder_buffer + 8;
 
-        cwraps.mono_wasm_string_get_data(mono_string, <any>ppChars, <any>pLengthBytes, <any>pIsInterned);
+        cwraps.mono_wasm_string_get_data(this.mono_wasm_string_root.value, <any>ppChars, <any>pLengthBytes, <any>pIsInterned);
```

Comment:
good catch, we should fix this on main immediately

Reply:
I plan to prototype in this PR and when we are happy about the prototype, I would take it apart into smaller PRs. This is one of them for sure. If you want to do it right away, feel free to do it.